.vsts-pipelines/builds/PRBuild.yml

```diff
@@ -0,0 +1,232 @@
+      $(PipebuildArgs)
+      /t:Prepare
+      /t:Restore
+      /t:GeneratePropsFiles
+      /t:BuildSharedFx
+      /p:SharedFxRID=win-x86
+      /p:BuildNumber=$(Build.BuildNumber)
+
+- template: ../templates/phases/default.yml
+  parameters:
+    name: SharedFxMusl
+    dependsOn: UniverseCoherence
+    beforeBuild:
+    - task: DownloadBuildArtifacts@0
+      displayName: Download artifacts
+      inputs:
+        artifactName: UniverseCoherence
+        downloadPath: $(Build.SourcesDirectory)/.deps
+    # Workaround https://github.com/Microsoft/vsts-tasks/issues/6739
+    - task: CopyFiles@2
+      displayName: Copy package assets to correct folder
+      inputs:
+        sourceFolder: $(Build.SourcesDirectory)/.deps/UniverseCoherence
+        targetFolder: $(Build.SourcesDirectory)/.deps
+    agentOs: Linux
+    buildScript: dockerbuild.sh
+    buildArgs: >
+      alpine
+      $(PipebuildArgs)
+      /t:Prepare
+      /t:GeneratePropsFiles
+      /t:BuildSharedFx
+      /p:SharedFxRID=linux-musl-x64
+      /p:BuildNumber=$(Build.BuildNumber)
+#END SharedFx
+
+# BEGIN Tests
+- template: ../templates/phases/default.yml
+  parameters:
+    name: Test_Windows
+    timeoutInMinutes: 180
+    dependsOn: UniverseCoherence
+    beforeBuild:
+    - task: DownloadBuildArtifacts@0
+      displayName: Download artifacts
+      inputs:
+        artifactName: UniverseCoherence
+        downloadPath: $(Build.SourcesDirectory)/.deps
+    # Workaround https://github.com/Microsoft/vsts-tasks/issues/6739
+    - task: CopyFiles@2
+      displayName: Copy package assets to correct folder
+      inputs:
+        sourceFolder: $(Build.SourcesDirectory)/.deps/UniverseCoherence
+        targetFolder: $(Build.SourcesDirectory)/.deps
+    - task: NodeTool@0
+      displayName: Use Node 8.x
+      inputs:
+        versionSpec: 8.x
+    buildArgs: >
+      $(PipebuildArgs)
+      /p:Configuration=Release
+      /p:TestOnly=true
+      /p:VSTestVerboseOutput=true
+    hasTests: true
+
+- template: ../templates/phases/default.yml
+  parameters:
+    name: Test_Linux
+    timeoutInMinuts: 120
+    dependsOn: UniverseCoherence
+    beforeBuild:
+    - task: DownloadBuildArtifacts@0
+      displayName: Download artifacts
+      inputs:
+        artifactName: UniverseCoherence
+        downloadPath: $(Build.SourcesDirectory)/.deps
+    # Workaround https://github.com/Microsoft/vsts-tasks/issues/6739
+    - task: CopyFiles@2
+      displayName: Copy package assets to correct folder
+      inputs:
+        sourceFolder: $(Build.SourcesDirectory)/.deps/UniverseCoherence
+        targetFolder: $(Build.SourcesDirectory)/.deps
+    - task: NodeTool@0
+      displayName: Use Node 8.x
+      inputs:
+        versionSpec: 8.x
+    agentOs: Linux-Big
+    buildScript: build.sh
+    buildArgs: >
+      $(PipebuildArgs)
+      /p:Configuration=Release
+      /p:TestOnly=true
+      /p:VSTestVerboseOutput=true
+    hasTests: true
+
+- template: ../templates/phases/default.yml
```

Comment:
Lots of redundancy in the test steps. Looks like a good candiate for a template.